libs/R-TypeLogic/Server/Systems/DeathLifeSystem.hpp

```diff
@@ -22,9 +24,18 @@ namespace ecs
         {
             std::vector<std::shared_ptr<ecs::Entity>> joined = world.joinEntities<Life>();
 
-            auto deathlifetime = [](std::shared_ptr<ecs::Entity> entityPtr) {
+            auto deathlifetime = [&world](std::shared_ptr<ecs::Entity> entityPtr) {
                 if (entityPtr.get()->getComponent<Life>().lifePoint <= 0) {
-                    entityPtr.get()->addComponent<Death>();
+                    if (entityPtr->contains<Player>()) {
+                        Client client = world.getTransisthorBridge()->getCommunicatorInstance().getClientByHisId(
```

Comment:
Why ?

Reply:
Refering to the PR description : "Now, the room instance send a protocol 13 to the client to kick him from the game. (In future, we can pass data with protocol 13 to specify which type of kick is it)" -> In case of a player death, i use the Disconnectable component and not the Death one. (Tell to the client the game is ended for him, and clean the player data from the room)